### ISSUE TITLE
FR: Squish Square

### ISSUE THREAD

**Issue by PhilBaz** [5axes/SquishSquare#1]
I've started including 2 30mm x 30mm x .31mm squares in every print. I use 'layer start' X & Y to ensure they are printed first. I use these to judge and adjust first layer squish. Better than starting the print over.

I have an enclosed printer and do lots of ABS. With thermal expansion constant revaluation is require for back to back prints....unfortunately.

Would love to have these shapes included in your plug-in.  I think its very useful. I currently import them for every print, as its faster than resizing a cube.

![Screenshot 2022-06-17 215802](https://user-images.githubusercontent.com/21558241/174421955-2fc801fd-3c35-46da-88c9-ebf9d118a053.jpg)

**Comment by PhilBaz** [comment]
@5axes Awesome!!! Much appreciated!

**Comment by PhilBaz** [comment]
@5axes 

Just updated by uncertain how to get the result you have here?

**Comment by PhilBaz** [comment]
@5axes 

Ohhh, its its own plug-in!! thought it was integrated into Anti-tab+ because of the git lables..... Sorry,

Thank you !!!!!

**Comment by 5axes** [comment]
It's not the same goal as the tab solution. So for me it was more logical  to create a new plugin for a new function.  Right now this plugin is more a prototype , I will not release it . Perhaps some people with similar needs will give some feedback to improve this solution.

**Comment by PhilBaz** [comment]
Understood...

Based on my original request, it works very well. I love the automatic 2 square creation, and that it changes the pattern type. Also, when more squares are created the patter type continues to alternate, that's fantastic! I'm really going to appreciate having this.

I would spread the word by I'm not active on social media. I frequent the Voron discord, but they tend to lean toward SuperSlicer.

The reason this tool is so great is because with an enclosed printers you get thermal expansion of the frame and gantry, that changes the z-offset, so it must be constantly tuned. So the crowd who will benefit from this most is a crowd with enclosed printers.

**Comment by 5axes** [comment]
As improvement it would be great to add a square tab in the four corner of the part. But as we cannot manage this order in Cura, we must use a postprocessing script for that. Just one Idea 
One other modification right now We have 3 patterns in Cura. Right now zigzag is not set any advantage to use also this pattern ?

**Comment by PhilBaz** [comment]
@5axes 

Yes, the four corner might be nice for those who don't use mesh bed leveling. I use Mesh bed so Z-offset wanders as the machine heats, but is relatively consistent across the bed.

Auto mode currently creates two squares per item. It might be nice to have it just make two squares. Some times I print plates full of parts and still just need the two squares to tune the global z-offset.

I don't find the difference between zig-zag and lines to reveal any information. However, if too close the concentric square will start to go wavy as it goes inward, as the corner radius gets smaller and smaller.  Lines/zig-zag are less sensitive and wont go wavy at the same height as concentric. Though they will eventually go wavy if too close.

Thanks again!!! Already using it on every print and loving it!

**Comment by 5axes** [comment]
If you select one element and use the automatique function , just two square will be create on the builtplate

**Comment by PhilBaz** [comment]
@5axes 

Humm. I tried and still got a bunch of squares. I had some parts grouped. But I tried a few different ways and still always got a bunch. 

Ill try again and attach file if I can reproduce it.

**Comment by PhilBaz** [comment]
@5axes 

Ahh, okay. Thought I might have done something wrong. 

If you get to 2.0 might be nice to have a check box for 'all' to toggle the function from a selected object to all objects.

**Comment by 5axes** [comment]
No but create two tabs directly in the individual creation, could be  a solution

**Comment by PhilBaz** [comment]
@5axes  Hey, just saw this... Ill try it and report back.

I also noticed the keyboard shortcut for the plugin is 's' and it steals this shortcut from the scale function. Can it be changed to Shift+s ?

**Comment by PhilBaz** [comment]
@5axes Humm...I added the line, but I'm not sure how the behaviour changed.

What is the expected change in behaviour?

 The auto button still adds 2 squares for every part... And, if I select a part I get one square for every time I click on the part--when it is selected and the squish square panel is open.

**Comment by 5axes** [comment]
The non automatic function (when you clic on a part) should now add two tabs (or the number you have define for this value)

**Comment by PhilBaz** [comment]
@5axes 

Ahh, I just realized I had to update the plugin as well. Yes, its working perfectly. Fantastic!

**Comment by PhilBaz** [comment]
Is the color of the buttons set by my system or by Cura? Its difficult to see the text. 

Its the same for tab+.

Im on Cura 5.
![Screenshot 2022-07-10 122556](https://user-images.githubusercontent.com/21558241/178157413-066f2e9a-6533-4888-8500-11cc56890e8d.jpg)
